From e9b25c7c040d8526fe8675b97d1067c8ffa7249f Mon Sep 17 00:00:00 2001
From: David Faure <faure@kde.org>
Date: Tue, 16 Aug 2016 15:50:57 +0200
Subject: [PATCH] Fix wrong value of __KDE_HAVE_GCC_VISIBILITY on systems with
 gcc 6.

Clearly the author of this regexp, in 2006, thought gcc would never
hit version 6 :-)

REVIEW: 128697
---
 cmake/modules/FindKDE4Internal.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/modules/FindKDE4Internal.cmake b/cmake/modules/FindKDE4Internal.cmake
index 7d54b9b..c345eca 100644
--- a/cmake/modules/FindKDE4Internal.cmake
+++ b/cmake/modules/FindKDE4Internal.cmake
@@ -1221,7 +1221,7 @@ if (CMAKE_COMPILER_IS_GNUCXX)
    # get the gcc version
    exec_program(${CMAKE_C_COMPILER} ARGS ${CMAKE_C_COMPILER_ARG1} --version OUTPUT_VARIABLE _gcc_version_info)
 
-   string (REGEX MATCH "[345]\\.[0-9]\\.[0-9]" _gcc_version "${_gcc_version_info}")
+   string (REGEX MATCH "[3-9]\\.[0-9]\\.[0-9]" _gcc_version "${_gcc_version_info}")
    # gcc on mac just reports: "gcc (GCC) 3.3 20030304 ..." without the patch level, handle this here:
    if (NOT _gcc_version)
       string (REGEX MATCH ".*\\(GCC\\).* ([34]\\.[0-9]) .*" "\\1.0" _gcc_version "${gcc_on_macos}")
-- 
2.8.1

